internal/server/files.go

```diff
@@ -0,0 +1,190 @@
+package server
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/gofrs/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+type files struct{}
+
+type Resource struct {
+	RemoteIdentifier string `json:"remoteIdentifier"`
+}
+
+func respond(context echo.Context, code int, message string) error {
+	success := code < 300
+	return context.JSON(code, echo.Map{
+		"success": success,
+		"message": message,
+	})
+}
+
+func readValetToken(context echo.Context) (*ValetToken, error) {
+	valetTokenBase64 := context.Request().Header.Get("x-valet-token")
+	valetTokenBytes, err := base64.StdEncoding.DecodeString(valetTokenBase64)
+	if err != nil {
+		return nil, errors.New("Unable to parse base64 valet token")
+	}
+	valetTokenJson := string(valetTokenBytes)
+	var token ValetToken
+	if err := json.Unmarshal([]byte(valetTokenJson), &token); err != nil {
+		return nil, errors.New("Unable to parse json valet token")
+	}
+	return &token, nil
+}
+
+type ValetRequestParams struct {
+	Operation string `json:"operation"`
+	Resources []Resource
+}
+
+type ValetToken struct {
+	Authorization string `json:"authorization"`
+	FileID        string `json:"fileId"`
+}
+
+func (token *ValetToken) getFilePath() (*string, error) {
+	id, err := uuid.FromString(token.FileID)
+	if err != nil {
+		return nil, errors.New("Unable to parse json valet token")
+	} else if !fs.ValidPath(id.String()) {
+		return nil, errors.New("Invalid path")
+	}
+	// TODO: Allow custom path in config
+	// TODO: Subfolders for each user (Compatible format with official server)
```

Comment:
go1.24 added https://pkg.go.dev/os#Root which helps to improve filesystem security.